File: Exercism Codes/PHP/AtbashCipher.php
<?php

declare(strict_types=1);

function encode(string $text): string
{
    $result = [];
    foreach (str_split($text) as $x) {
        if (preg_match("/\w/", $x)) {
            $result[] = transform($x);

            if ((count($result) + 1) % 6 === 0) {
                $result[] = " ";
            }
        }
    }
    return trim(implode("", $result));
}

function decode(string $text): string
{
    $result = [];
    foreach (str_split($text) as $x) {
        if (preg_match("/\w/", $x)) {
            $result[] = transform($x);
        }
    }
    return implode("", $result);
}

function transform(string $x): string
{
    return preg_match("/[A-Za-z]/", $x) ? mb_chr((26 - (mb_ord(strtolower($x)) - 97) - 1) + 97) : $x;
}

File: Exercism Codes/Python/atbash_cipher.py
def encode(plain_text):
    result = []
    for c in plain_text:
        if c.isalnum():
            result.append(transform(c))

            if (len(result) + 1) % 6 == 0:
                result.append(" ")

    return "".join(result).strip()


def decode(ciphered_text):
    return "".join(transform(x) for x in ciphered_text if x.isalnum())

def transform(x):
    return chr((26 - (ord(x.lower()) - 97) - 1) + 97) if x.isalpha() else x